src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java

```diff
@@ -21,13 +24,16 @@
 
     private PathMapper pathMapper = new PathMapper();
 
-    public ExecutionResultNode mapERNFromUnderlyingToOverall(ExecutionResultNode node, UnapplyEnvironment environment) {
+    public ExecutionResultNode mapERNFromUnderlyingToOverall(ExecutionResultNode node, UnapplyEnvironment environment, AtomicInteger typeRenameCount) {
 
         Map<String, String> typeRenameMappings = environment.typeRenameMappings;
         GraphQLSchema overallSchema = environment.overallSchema;
         ExecutionPath mappedPath = pathMapper.mapPath(node.getExecutionPath(), node.getResultKey(), environment);
         GraphQLObjectType mappedObjectType = mapObjectType(node, typeRenameMappings, overallSchema);
         GraphQLFieldDefinition mappedFieldDefinition = getFieldDef(overallSchema, mappedObjectType, node.getFieldName());
+
+        int typeDecrementValue = node instanceof ListExecutionResultNode ? - node.getChildren().size() : 0;
```

Comment:
did you mean `negative node.getChildren().size()` here?

Reply:
yep this should be negative, i'll fix the spacing